core/src/test/scala/org/apache/spark/sql/delta/stats/DataSkippingDeltaTests.scala

```diff
@@ -539,18 +545,23 @@ trait DataSkippingDeltaTestsBase extends QueryTest
     hits = Seq(
       "a IS NULL",
       "a = NULL",  // Ideally this should not hit as it is always FALSE, but its correct to not skip
-      "a <=> NULL",
+      "NOT a = NULL", // Same as previous case
+      "a <=> NULL", // This is optimized to `IsNull(a)` by NullPropagation
       "TRUE",
       "FALSE",     // Ideally this should not hit, but its correct to not skip
       "NULL AND a = 1" // This is optimized to FALSE by ReplaceNullWithFalse, so it's same as above
     ),
     misses = Seq(
       // stats tell us a is always NULL, so any predicate that requires non-NULL a should skip
       "a IS NOT NULL",
+      "NOT a <=> NULL", // This is optimized to `IsNotNull(a)`
       "a = 1",
+      "NOT a = 1",
       "a > 1",
       "a < 1",
-      "a <> 1"
+      "a <> 1",
+      "a <=> 1",
+      "NOT a <=> 1"
```

Comment:
Is this a bug? When `a` is `null`, `a <=> 1` should be `false` and `NOT a <=> 1` should be `true`.